### ISSUE TITLE
git branching workflow for use_release_issue()?

### ISSUE THREAD

**Issue by kevinushey** [r-lib/usethis#1884]
Using https://github.com/rstudio/renv/issues/1634 as a template, I wonder if we should also suggest starting with something like:

- [ ] Create and switch to release branch `release/<version>`

Then, follow the regular steps. After building and submitting to CRAN, if the submission is accepted, we can then:

- [ ] Merge release branch to main

And then run the regular steps. Does that make sense?

**Comment by gadenbuie** [comment]
A common thread between this and #1887 in my work is that I would appreciate the ability to customize the release issue in a more detailed way. Currently, `release_bullets()` adds steps to the end of "Prepare for release", but a branching workflow or a workflow that require additional checks might benefit from customizing where the todo items appear in the list.

One option would be to allow `release_bullets()` to return a named list such that items can be added to specific places in the release checklist, e.g. `"prepare-start"`, `"pre-check"`, `"post-check"`, etc. This could be backwards compatible in that an unnamed list would be assumed to be `"prepare-end"` (at the end of the prepare section).

If there's interest in this approach, I'd be happy to work up a PR.

**Comment by hadley** [comment]
I'd be happy to make this the default workflow, assuming we can automate the steps sufficiently that I can follow the checklist without having to think about it. Maybe it's sufficient to add `pr_init()`, `pr_push()`, and `pr_finish()` in the appropriate places? I think it would be useful with a concrete proposal there.

**Comment by DavisVaughan** [comment]
@hadley I have done that actually. It is probably a little out of date now but I did it here https://github.com/business-science/tibbletime/issues/104, so we could maybe pull that over?

**Comment by hadley** [comment]
Yeah, that seems like a good start. Want to do a PR?  (I'd leave the change in revdeps for now since we should discuss that elsewhere)

**Comment by hadley** [comment]
More discussion in #1946
